### PR TITLE
refactor bot sample

### PR DESCRIPTION
**Pull request by a1exwang**

### CODE REVIEW

File: bot/api.ts [hunk 1]
```diff
@@ -0,0 +1,45 @@
+// Import required packages
```
Comment (kimizhu) [inline bot/api.ts]: Is it better to name this file as index.ts? As we want it general and it's not necessary only HTTP trigger.

File: bot/api.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+});
+
+// Create conversation reference storage
+const conversationReferenceStore = new ConversationReferenceStore();
+// Create the bot that will handle incoming messages.
+const bot = new TeamsBot(conversationReferenceStore);
+// Create notification sender to proactively send outgoing messages.
+const notificationSender = new NotificationSender(adapter);
+
+// Listen for incoming requests.
```
Comment (kimizhu) [inline bot/api.ts]: Bot listens for incoming requests

File: bot/api.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+server.post("/api/messages", async (req, res) => {
+  return await adapter.processActivity(req, res, async (context) => {
+    await bot.run(context);
+  });
+});
+
+// Send notification
```
Comment (kimizhu) [inline bot/api.ts]: Explicitly comment it is HTTP trigger.

File: bot/api.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+server.post("/api/notification", async (req, res) => {
+  await notificationSender.sendNotification(
+    conversationReferenceStore.get(),
+    "Hello world!\nYou've received a notification triggered by API."
```
Comment (kimizhu) [inline bot/api.ts]: What is this for?
Can we change it to something like: context.log('HTTP trigger function processed a request.');

File: bot/store.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+export class ConversationReferenceStore {
+  _ref: Partial<ConversationReference> | undefined
+
+  get(): Partial<ConversationReference> | undefined {
+    return this._ref;
+  }
+
+  set(conversationReference: Partial<ConversationReference>) {
+      // You can persist the conversationReference to a file, database, etc. to pro-actively send messages at any time.
```
Comment (kimizhu) [inline bot/store.ts:11]: Is this comment still valid? At least it should out of this setter.

**Comment by swatDong** [comment]
For the arch figure:
- The `url === "xxx"` condition looks weird in `1.2` and `2.2`, better to separate the arrow points.
- A tiny improvement is to use different colors for the two major flows.

E.g., 
image

File: bot/api.ts (hunk 1 above)
Comment (a1exwang) [reply bot/api.ts]: Changed

File: bot/api.ts (hunk 3 above)
Comment (a1exwang) [reply bot/api.ts]: updated

File: bot/api.ts (hunk 2 above)
Comment (a1exwang) [reply bot/api.ts]: updated

**Review by kimizhu — APPROVED** [review]

**Pull request merged**